src/engine/sparqlExpressions/LiteralExpression.h

```diff
@@ -224,4 +225,38 @@ using IdExpression = detail::LiteralExpression<ValueId>;
 using VectorIdExpression =
     detail::LiteralExpression<VectorWithMemoryLimit<ValueId>>;
 using SingleUseExpression = detail::SingleUseExpression;
+
+namespace detail {
+
+//______________________________________________________________________________
+using IdOrLocalVocabEntry = prefilterExpressions::IdOrLocalVocabEntry;
+// Given a `SparqlExpression*` pointing to a `LiteralExpression`, this helper
+// function retrieves a corresponding `IdOrLocalVocabEntry` variant
+// (`std::variant<ValueId, LocalVocabEntry>`) for `LiteralExpression`s that
+// contain a suitable type.
+// Given the boolean flag `stringAndIriOnly` is set to `true`, only `Literal`s,
+// `Iri`s and `ValueId`s of type `VocabIndex`/`LocalVocabIndex` are returned. If
+// `stringAndIriOnly` is set to `false` (default), all `ValueId` types retrieved
+// from `LiteralExpression<ValueId>` will be returned.
+inline std::optional<IdOrLocalVocabEntry>
+getIdOrLocalVocabEntryFromLiteralExpression(const SparqlExpression* child,
+                                            bool stringAndIriOnly = false) {
+  using enum Datatype;
+  if (const auto* idExpr = dynamic_cast<const IdExpression*>(child)) {
+    auto idType = idExpr->value().getDatatype();
+    if (stringAndIriOnly && idType != VocabIndex && idType != LocalVocabIndex) {
+      return std::nullopt;
+    }
```

Comment:
This case is testable (directly for this function as well as indirectly for the `StrStarts` function, choose one.